src/server/frontend_xwayland/xwayland_wm.cpp

```diff
@@ -76,6 +76,23 @@ static const struct cursor_alternatives cursors[] = {
 
 namespace mf = mir::frontend;
 
+namespace
+{
+template<typename T>
+auto c_array_to_string(T* data, size_t size) -> std::string
+{
+    std::string result = "[";
+    for (unsigned i = 0; i < size; i++)
+    {
+        if (i)
+            result += ", ";
+        result += std::to_string(*(data + i));
+    }
+    result += "]";
```

Comment:
If you're going to index into the "array", use `data[i]`.

If you're going to dereference pointers, use pointers! (`for (T* p = data; p != data+size; ++p)`)

Reply:
Done